### PR TITLE
`MediaGallerySection` Component

### PR DESCRIPTION
**Pull request by ScottyMJacobson**
Spec and conversation https://www.notion.so/stackbit/stackbit-components-create-media-gallery-component-ef6a897d9301442a9384f9c51ffbd8e9

Basic feature overview:

- Basic section features / styling
- Grid of `ImageBlock`s
- Configurable images-per-row
- Configurable max image size
- Caption and hover features

### CODE REVIEW

File: src/components/MediaGallerySection/media-gallery-section.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            control: { type: 'select' }
+        },
+        elementId: {}
+    }
+};
+
+const Template = (args: MediaGallerySectionProps) => <MediaGallerySection {...args} />;
+
+const args: MediaGallerySectionProps = {
+    title: 'Some photos',
```
Comment (JugglerX) [inline src/components/MediaGallerySection/media-gallery-section.stories.tsx:31]: @smnh @eladroz. These default values should ideally come direct from the model to avoid repetition and de-syncing content. The values shown in the story should be the same as the model defaults and as you can see here the values are already diverging.

@ScottyMJacobson no need to fix this. Just wanted to highlight it to Simon for a possible future improvement.

**Review by JugglerX — CHANGES_REQUESTED** [review]

File: models/MediaGallerySection.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    name: subtitle
+    label: Subtitle
+    default: Aren't they lovely photos?
+  - type: list
+    name: images
+    label: Images
+    items:
+      type: model
+      models: [ImageBlock]
+      # TODO - default
```
[the remark this answers is not in the document]
Comment (TomasBankauskas) [reply models/MediaGallerySection.yaml:29]: @JugglerX yes we support and should provide defaults.

[the remark this answers is not in the document]
Comment (ScottyMJacobson) [reply models/MediaGallerySection.yaml:29]: Should the default be a hard-coded object in the yaml, or a reference to a data .json (which, i think, would then have to be included in themes?).

File: src/components/MediaGallerySection/media-gallery-section.stories.tsx (hunk 1 above)
Comment (ScottyMJacobson) [reply src/components/MediaGallerySection/media-gallery-section.stories.tsx:26]: That makes sense! I mean - we could parse the YAML and pass the defaults into the args (or use the same stackbit innards to do it), right? It's all just a JavaScript runtime

_(This goes back to my joking "singularity" comment over slack about how storybook and stackbit studio aren't so far apart after all...)_

File: models/MediaGallerySection.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScottyMJacobson) [reply models/MediaGallerySection.yaml:29]: Also - what's the best way to test defaults? I'm gonna create an "empty" preset (empty data prop) and create a new section with it, but if there's a better way let me know!

[the remark this answers is not in the document]
Comment (ScottyMJacobson) [reply models/MediaGallerySection.yaml:29]: Here's what I'm trying:

```
  - type: list
    name: images
    label: Images
    items:
      type: model
      models: [ImageBlock]
      default:
        - type: ImageBlock
          url: https://loremflickr.com/cache/resized/65535_51377543930_d101570b6d_b_640_640_nofilter.jpg
          altText: Cute Cat
          caption: Look! It's a cat!
```

But when I make a new section using my "empty" preset, I get the default values for everything else but not for `images` (it's an empty array) 🤔 

_(also tried it without the `type` field - but was trying to match what ends up in the content after adding an image in the GUI)_

**Review by TomasBankauskas — APPROVED** [review]
Looks good, thank you @ScottyMJacobson.

**Review by TomasBankauskas — APPROVED** [review]
